packages/bot/src/features/avatar.ts

```diff
@@ -0,0 +1,153 @@
+	): Promise<void> {
+		const embed = getEmbeds(interaction.message, "user");
+		if (!embed)
+			throw new BotError(this.ul("error.embed.notFound"), {
+				cause: "AVATAR_EDIT",
+				level: BotErrorLevel.Warning,
+			});
+		const jsonEmbed = embed.toJSON().thumbnail?.url;
+		let thumbnail = jsonEmbed
+			? cleanAvatarUrl(jsonEmbed)
+			: await fetchAvatarUrl(interaction.guild!, interaction.user);
+		if (thumbnail.match(QUERY_URL_PATTERNS.DISCORD_CDN)) thumbnail = "";
+		const modal = new Djs.ModalBuilder()
+			.setCustomId("editAvatar")
+			.setTitle(this.ul("button.avatar.description"))
+			.addLabelComponents((label) =>
+				label
+					.setLabel(this.ul("modals.avatar.name"))
+					.setDescription(this.ul("modals.avatar.description"))
+					.setTextInputComponent((input) =>
+						input
+							.setCustomId("avatar")
+							.setValue(thumbnail)
+							.setRequired(false)
+							.setStyle(Djs.TextInputStyle.Short)
+					)
+			)
+			.addLabelComponents((label) =>
+				label
+					.setLabel(this.ul("modals.avatar.name"))
+					.setDescription(this.ul("modals.avatar.file.description"))
+					.setFileUploadComponent((file) =>
+						file.setCustomId("avatarFile").setRequired(false).setMaxValues(1)
+					)
+			);
+
+		await interaction.showModal(modal);
+	}
+
+	/**
+	 * Handles a Discord modal submission to update a user's avatar in an embed message.
+	 *
+	 * Validates the provided avatar URL, updates the embed's thumbnail if valid, edits the original message with the new embed, and sends an ephemeral confirmation reply to the user.
+	 *
+	 * @throws {Error} If the user embed is not found in the message.
+	 */
+	async edit(): Promise<void> {
+		const interaction = this.interaction as Djs.ModalSubmitInteraction;
+		if (!interaction.message) return;
+		profiler.startProfiler();
+		const message = await (interaction.channel as TextChannel).messages.fetch(
+			interaction.message.id
+		);
+		await interaction.deferReply({ flags: Djs.MessageFlags.Ephemeral });
+		const files: Djs.AttachmentBuilder[] = [];
+		const uploaded = interaction.fields.getUploadedFiles("avatarFile")?.first();
+		let avatar = "";
+
+		if (uploaded) {
+			if (!uploaded.contentType?.match(QUERY_URL_PATTERNS.VALID_EXTENSIONS))
+				return await reply(interaction, {
+					embeds: [embedError(this.ul("error.avatar.format"), this.ul)],
+				});
+			const attachment = new Djs.AttachmentBuilder(uploaded.url, { name: uploaded.name });
+			files.push(attachment);
+			avatar = `attachment://${attachment.name}`;
+		} else {
+			const input = cleanAvatarUrl(interaction.fields.getTextInputValue("avatar"));
+			if (!input)
+				return await reply(interaction, {
+					embeds: [embedError(this.ul("error.avatar.missing"), this.ul)],
+				});
+			if (input.match(QUERY_URL_PATTERNS.DISCORD_CDN))
+				return await reply(interaction, {
+					embeds: [embedError(this.ul("error.avatar.cdn"), this.ul)],
+				});
+			if (!verifyAvatarUrl(input))
+				return await reply(interaction, {
```

Comment:
TS2322: Type
Message<boolean> | InteractionResponse<boolean>
is not assignable to type void
Type Message<boolean> is not assignable to type void